app/javascript/packs/application.js

```diff
@@ -1,7 +1,9 @@
 import main from '../mastodon/main';
 
 if (!window.Intl || !Object.assign || !Number.isNaN ||
-    !window.Symbol || !Array.prototype.includes) {
+    !window.Symbol || !Array.prototype.includes ||
+    !window.IntersectionObserver ||
+    !window.requestIdleCallback) {
```

Comment:
I love both `IntersectionObserver` and `requestIdleCallback`, but I'm not sure if I'm comfortable with `polyfills.js` becoming a "not Chrome" tax. In current master, the latest version of Chrome/Edge/Firefox/Safari don't need to load the polyfill; it's only e.g. iOS 9 that needs it. Whereas `IntersectionObserver` is only in Chrome and Edge, and `requestIdleCallback` is only in Chrome and Firefox.

I would prefer to create a `polyfills2.js` or something, so that we can differentiate polyfills that are 1) are in the latest version of each major browser vs 2) in select versions of Chrome/Firefox/Edge.